gtests/net/mptcp/sockopts/sockopt_cork_multi.pkt

```diff
@@ -0,0 +1,50 @@
++0.0  fcntl(3, F_GETFL) = 0x2 (flags O_RDWR)
++0.0  fcntl(3, F_SETFL, O_RDWR|O_NONBLOCK) = 0
+
++0    bind(3, ..., ...) = 0
++0    listen(3, 1) = 0
+
++0.0    <  addr[caddr0] > addr[saddr0]  S   0:0(0)               win 65535  <mss 1460, sackOK, TS val 4074410674 ecr 0,          nop, wscale 8, mpcapable v1 flags[flag_h] nokey>
++0.0    >                               S.  0:0(0)        ack 1             <mss 1460, sackOK, TS val 4074410674 ecr 4074410674, nop, wscale 8, mpcapable v1 flags[flag_h] key[skey]>
++0.0    <                                .  1:1(0)        ack 1  win 256    <nop, nop,         TS val 4074410674 ecr 4074410674,                mpcapable v1 flags[flag_h] key[ckey=2, skey]>
++0    accept(3, ..., ...) = 4
+
++0.2    <                               P.  1:3(2)        ack 1  win 256    <nop, nop, TS val 4074418292 ecr 4074410674,                        mpcapable v1 flags[flag_h] key[skey, ckey] mpcdatalen 2, nop, nop>
+// MP_CAPABLE carrying data are acked with 64-bit, safer not knowing if the
+// sender will use a DSN on 64-bit or 32-bit. Later we can use the lower 32 bits
++0.0    >                                .  1:1(0)        ack 3             <nop, nop, TS val 4074418293 ecr 4074418292, dss dack8=3  dll=0 nocs>
++0.0  read(4, ..., 2) = 2
+
+// more inbound data and MP_PRIO to flip active -> backup
++0.0    <                               P.  3:5(2)        ack 1  win 256                                                <dss dack8=1  dsn4=3 ssn=3 dll=2  nocs, mp_prio backup=1, nop, nop, nop>
++0.0    >                                .  1:1(0)        ack 5             <nop, nop, TS val 4074418293 ecr 4074418292, dss dack4=5  nocs>
++0.0  read(4, ..., 2) = 2
+
+// create subflow
++0.0    <  addr[caddr1] > addr[saddr0]  S   0:0(0)               win 65535  <mss 1460, sackOK, TS val 448955294 ecr 0,         nop, wscale 8, mp_join_syn     backup=1 address_id=1 token=sha256_32(skey)>
++0.0    >                               S.  0:0(0)        ack 1             <mss 1460, sackOK, TS val 448955294 ecr 448955294, nop, wscale 8, mp_join_syn_ack backup=1 address_id=0 sender_hmac=auto>
++0.0    <                                .  1:1(0)        ack 1  win 256    <nop, nop, TS val 448955294 ecr 448955294,                        mp_join_ack sender_hmac=auto>
++0.0    >                                .  1:1(0)        ack 1             <nop, nop, TS val 448955294 ecr 448955294,                        dss dack4=5 nocs>
+
+// more inbound data and MP_PRIO to flip backup -> active
++0.0    <                               P.  1:11(10)      ack 1   win 256                                               <dss dack8=1  dsn4=5 ssn=1 dll=10 nocs, mp_prio backup=0, nop, nop, nop>
++0.0    >                                .  1:1(0)        ack 11            <nop, nop, TS val 448955294 ecr 448955294,   dss dack4=15 nocs>
++0.0  read(4, ..., 10) = 10
+
+// check that TCP_CORK was cloned from listener
++0.0 getsockopt(4, SOL_TCP, TCP_CORK,  [1], [4]) = 0
```

Comment:
if you move this `getsockopt()` call to the test with a single flow, do you think we still need this "multi" test?

I mean: that's good to check with multiple subflows but I'm not sure we are covering more code with it. Or do we?

Reply:
I really wanted for the subflow loop in setsockopt to spin a couple of times, just to double-check that I have correct locking and change the right sockets. I also was toying a bit with packetdrill to gain some understanding. Now I agree that this doesn't add any meaningful coverage, thank you for pointing this out. WIll remove the multi test.

Reply:
:+1: Good idea to play a bit with Packetdrill. This is a nice tool but not always easy to use :)